core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -472,12 +469,19 @@ private SyncStatus syncInternal(long syncStartTs) throws
       // process the sync result for the original path
       try {
         path.traverse();
+        if (path.fullPathExists()) {
+          rootPathIsFile = !path.getInode().isDirectory();
+        }
       } catch (InvalidPathException e) {
         updateMetrics(false, startTime, syncPathCount, failedSyncPathCount);
         throw new RuntimeException(e);
       }
+    } catch (FileDoesNotExistException e) {
+      LOG.warn("Failed to sync metadata on root path {} due to FileDoesNotExistException",
```

Comment:
Actually I added this here because before it was printing out the whole stack trace (I changed it to only print the one line message) since it was caught in the catch all a couple lines below and was spamming the logs.

This can also happen on the first time you add a file to Alluxio and you have not synced the path yet, so it checks the UFS to see if the file exists before adding it to Alluxio.

I wouldn't mind setting this as a debug log tbh. What do you think?